src/api/java/com/minecolonies/api/colony/buildings/IBuilding.java

```diff
@@ -32,6 +33,27 @@
 
 public interface IBuilding extends IBuildingContainer, IRequestResolverProvider, IRequester
 {
+    /**
+     * Check if the building has a particular module.
+     * @param clazz the module of the class to check.
+     * @return true if so.
+     */
+    boolean hasModule(final Class<? extends IBuildingModule> clazz);
+
+    /**
+     * Get the module with a particular class.
+     * @param clazz the modules class.
+     * @return the module or null of not existant.
+     */
+    @Nullable
+    <T extends IBuildingModule> T getModule(Class<T> clazz);
+
+    /**
+     * Register a specific module to the building.
+     * @param module the module to register.
+     */
+    void registerModule(@NotNull final IBuildingModule module);
```

Comment:
I do not think this method should be in the API.
As stated before: building modules should be immutable once the building is constructed.
So I do not see a reason to have a registerModule method.

Reply:
Because we handle IBuilding everywhere. As an alternative we could set up sth like "you can't add modules after the initial setup" with a throw or so.

Reply:
But we should have this in IBuilding such that we don't have to give out the AbstractBuilding or so

Reply:
hm this reminds me a bit on the cook, which through upgrading a building gets added new functionality in terms of the assistant

Reply:
In that case the building would get the module form the beginning, but the module wouldn't "work" unless the necessary conditions are met.

Reply:
@Raycoms you should build the list of modules. When the building instance is constructed..,....
For example somewhere in the building type or something.

The building should never have this.

Reply:
We do this on building instantiation. But we give the building as a hook to register modules on. I don't see a problem with that.

Reply:
I can add to the javadoc: Must be called during building instantiation.